### PR TITLE
Add docker installation instructions (resolves #1281)

### LINKED ISSUE #1281: Build docs need to explain installing Docker for `make test`

### PR DESCRIPTION
**Pull request by jessebrennan**
resolves #1281

I added a separate area to specify Docker installation instructions. Let me know if this is too much or what I can do to improve!

### CODE REVIEW

File: docs/installation.rst [hunk 1]
```diff
@@ -238,6 +226,36 @@ this may come up empty. To fix it, run the following::
...
+
+       docker_registry=quay.io/USER make test
+
+   where ``USER`` is your Quay username.
+
+   For convenience you may want to add this variable to your bashrc by running
+
+   ::
+
+       echo "export export_registry=quay.io/USER" >> $HOME/.bashrc
```
Comment (hannes-ucsc) [inline docs/installation.rst]: `docker_registry`, not `export_registry`

Make sure you actually test every command before you paste them into documentation. Don't write them from memory.

**Comment by hannes-ucsc** [comment]
http://toil.readthedocs.io/en/issues-1281-installing-docker-docs/

File: docs/installation.rst [hunk 2]
```diff
@@ -238,6 +226,36 @@ this may come up empty. To fix it, run the following::
 
     for i in /usr/local/lib/python2.7/site-packages/*mesos*; do ln -snf $i venv/lib/python2.7/site-packages/ ; done
 
+.. _docker-note:
+.. topic:: Installing Docker
```
Comment (hannes-ucsc) [inline docs/installation.rst]: This treats Docker and Quay and the heading should reflect that. So should the sentence above that links to this note.

**Comment by hannes-ucsc** [comment]
Commit title is too long. Should stay below 76 characters.

**Comment by hannes-ucsc** [comment]
PR description should link to issue, too.

**Comment by hannes-ucsc** [comment]
… but I just added that reference.

File: docs/installation.rst (hunk 1 above)
Comment (jessebrennan) [inline docs/installation.rst]: I fixed this typo, thanks! When building the docs there are some warnings but none seem to be referring to code that I wrote (unless I'm missing something).

**Pull request merged**